### PR TITLE
audit improvements

### PR DESCRIPTION
**Pull request by adecaro**
ttx.audit: store transiant as soon as possible
zkatdlog driver: better logs

Linked to https://github.com/hyperledger-labs/fabric-smart-client/pull/453

Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: token/core/zkatdlog/nogh/auditor.go [hunk 1]
```diff
@@ -16,13 +16,15 @@ import (
 
 // AuditorCheck verifies if the passed tokenRequest matches the tokenRequestMetadata
 func (s *Service) AuditorCheck(tokenRequest *driver.TokenRequest, tokenRequestMetadata *driver.TokenRequestMetadata, txID string) error {
-	logger.Debugf("check token request validity...")
+	logger.Debugf("[%s] check token request validity, number of transfer action [%d]...", txID, len(tokenRequestMetadata.Transfers))
```
Comment (KElkhiyaoui) [inline token/core/zkatdlog/nogh/auditor.go:19]: number of transfer actions

File: token/core/zkatdlog/nogh/auditor.go [hunk 2]
```diff
@@ -16,13 +16,15 @@ import (
...
 	var inputTokens [][]*token.Token
-	for _, transfer := range tokenRequestMetadata.Transfers {
+	for i, transfer := range tokenRequestMetadata.Transfers {
+		logger.Debugf("[%s] transfer action [%d] contains [%d] inputs", txID, i, len(transfer.TokenIDs))
 		inputs, err := s.TokenCommitmentLoader.GetTokenOutputs(transfer.TokenIDs)
 		if err != nil {
 			return errors.Wrapf(err, "failed getting token outputs to perform auditor check")
 		}
+		logger.Debugf("[%s] transfer action [%d] contains [%d] inputs, loaded corresponding outputs [%d]", txID, i, len(transfer.TokenIDs), len(inputs))
```
Comment (KElkhiyaoui) [inline token/core/zkatdlog/nogh/auditor.go:27]: contains [%d] token IDs, loaded corresponding inputs [%d]

File: token/core/zkatdlog/nogh/sender.go [hunk 3]
```diff
@@ -110,11 +110,14 @@ func (s *Service) Transfer(txID string, wallet driver.OwnerWallet, ids []*token3
...
 	// audit info for senders
 	var senderAuditInfos [][]byte
-	for _, t := range tokens {
+	for i, t := range tokens {
 		auditInfo, err := htlc.GetOwnerAuditInfo(t.Owner, s)
 		if err != nil {
 			return nil, nil, errors.Wrapf(err, "failed getting audit info for sender identity [%s]", view.Identity(t.Owner).String())
 		}
+		if len(auditInfo) == 0 {
+			logger.Errorf("empty audit info for token [%s] ith owner [%s]", ids[i].String(), view.Identity(t.Owner))
```
Comment (KElkhiyaoui) [inline token/core/zkatdlog/nogh/sender.go:119]: for the owner [%s] of the i^th token [%s]

File: token/core/zkatdlog/nogh/auditor.go (hunk 1 above)
Comment (adecaro) [reply token/core/zkatdlog/nogh/auditor.go]: addressed

File: token/core/zkatdlog/nogh/auditor.go (hunk 2 above)
Comment (adecaro) [reply token/core/zkatdlog/nogh/auditor.go]: addressed

File: token/services/ttx/auditor.go [hunk 4]
```diff
@@ -99,10 +99,22 @@ func NewRegisterAuditorView(auditView view.View, opts ...token.ServiceOption) *R
 }
 
 func (r *RegisterAuditorView) Call(context view.Context) (interface{}, error) {
+	// register responder
 	if err := view2.GetRegistry(context).RegisterResponder(r.AuditView, &AuditingViewInitiator{}); err != nil {
 		return nil, errors.Wrapf(err, "failed to register auditor view")
 	}
-
+	// enable processing all token transactions for the given network and namespace
```
Comment (KElkhiyaoui) [inline token/services/ttx/auditor.go:106]: enable processing of all token transactions for the given...

File: token/services/ttx/auditor.go [hunk 5]
```diff
@@ -99,10 +99,22 @@ func NewRegisterAuditorView(auditView view.View, opts ...token.ServiceOption) *R
...
+	tms := token.GetManagementService(context, token.WithTMSID(r.TMSID))
+	if tms == nil {
+		return nil, errors.Errorf("cannot find tms for [%s]", r.TMSID)
+	}
+	net := network.GetInstance(context, tms.Network(), tms.Channel())
+	if tms == nil {
+		return nil, errors.Errorf("cannot find netowkr for [%s]", tms.ID())
```
Comment (KElkhiyaoui) [inline token/services/ttx/auditor.go:113]: cannot find network

**Pull request merged**